packages/video_player/video_player/lib/video_player.dart

```diff
@@ -499,7 +437,11 @@ class _VideoPlayerState extends State<VideoPlayer> {
 
   @override
   Widget build(BuildContext context) {
-    return _textureId == null ? Container() : Texture(textureId: _textureId);
+    return _textureId == null
+        ? Container()
+        : !kIsWeb
+            ? Texture(textureId: _textureId)
+            : HtmlElementView(viewType: _textureId.toString());
```

Comment:
@hterkelsen How can we avoid this implementation detail here? Or is it ok to leave it in for the moment?

Reply:
You could make the `Widget` built here be part of the platform interface

```dart
return _textureId == null ? Container() : platform.buildView(_textureId);
```

Reply:
Done